src/main/kotlin/com/ampnet/reportserviceeth/service/Extensions.kt

```diff
@@ -56,8 +56,13 @@ fun Long.toLocalDateTime(): LocalDateTime =
 fun BigInteger.toEther(): String = DecimalFormat(NUMBER_FORMAT)
     .format(Convert.fromWei(this.toBigDecimal(), Convert.Unit.ETHER))
 
-fun BigInteger.toMwei(): String = DecimalFormat(NUMBER_FORMAT)
-    .format(Convert.fromWei(this.toBigDecimal(), Convert.Unit.MWEI))
+fun BigInteger.formatWei(decimals: BigInteger): String = DecimalFormat(NUMBER_FORMAT)
+    .format(
+        Convert.fromWei(
+            this.toBigDecimal(),
+            Convert.Unit.values().first { it.weiFactor == decimals.toBigDecimal() }
```

Comment:
This can fail with `NoSuchElementException` - is this ok or do you maybe want to throw a more specific exception here?